Tasks/NuGetAuthenticateV1/main.ts

```diff
@@ -14,16 +17,36 @@ async function main(): Promise<void> {
         forceReinstallCredentialProvider = tl.getBoolInput("forceReinstallCredentialProvider", false);
         await installCredProviderToUserProfile(forceReinstallCredentialProvider);
 
-        // Configure the credential provider for both same-organization feeds and service connections
-        const serviceConnections = getPackagingServiceConnections('nuGetServiceConnections');
-        await configureCredProvider(ProtocolType.NuGet, serviceConnections);
+        var serviceConnections = [];
+
+        const feedUrl = tl.getInput("feedUrl");
+        const entraWifServiceConnectionName = tl.getInput("workloadIdentityServiceConnection");
+
+        if (feedUrl && entraWifServiceConnectionName)
+        {
+            // Is there a better way to enforce this? 
+            if (feedUrl.includes(",") || entraWifServiceConnectionName.includes(",")) {
+                throw new Error(tl.loc("MultipleValuesInSingleInput"));
+            }
+
+            await configureEntraCredProvider(ProtocolType.NuGet, feedUrl, entraWifServiceConnectionName);
+        }
+        else
+        {
+            // Configure the credential provider for both same-organization feeds and service connections
+            serviceConnections = getPackagingServiceConnections('nuGetServiceConnections');
+            await configureCredProvider(ProtocolType.NuGet, serviceConnections);
+        }
     } catch (error) {
-        tl.setResult(tl.TaskResult.Failed, error);
+        if (error.message.includes("existing service connection"))
```

Comment:
The original error message is being thrown using a localized string. I'm assuming the right thing to do here is to use a string with the same value and check for that instead?

Reply:
I think error type or field is more typical ? but I dont think typescript handles it as cleanly as c# for ex

Reply:
Zheng brought up the use of an error class, which is used in the common packaging code. This approach seems good to me, but it brought up another question [here](https://github.com/microsoft/azure-pipelines-tasks/pull/20246/files#r1704743358)

Reply:
I'll be deferring this to a following PR to keep this limited in scope to supporting WIF.